aws_protocol_adaptor/device_client/aws_nvmsgbroker.c

```diff
@@ -401,7 +401,7 @@ bool is_valid_connection_str(char *connection_str)
 	return true;
 }
 
-char *generate_sha256_hash(char *str)
+char *generate_sha256_hash(char *str, char *output_str)
```

Comment:
It's typical in C-style APIs for the destination variable to come first. See `sprintf` for an example.

Reply:
Also, you're passing in this argument `output_str` but you're not actually using it anywhere.